packages/engine/vscode/src/lib/editor.ts

```diff
@@ -1 +1,84 @@
-// Look at this : https://microsoft.github.io/language-server-protocol/specifications/specification-current/
\ No newline at end of file
+import { editorProfile, IEditor, Annotation, HighlightPosition } from '@remixproject/plugin-api';
+import { MethodApi } from '@remixproject/plugin-utils';
+import { window, Range, TextEditorDecorationType, Position, languages, DiagnosticCollection, Diagnostic, Uri, DiagnosticSeverity, TextEditor } from "vscode";
+import { PluginOptions } from "@remixproject/engine";
+import { CommandPlugin } from "./command";
+
+export interface EditorOptions extends PluginOptions {
+  language: string;
+}
+export class EditorPlugin extends CommandPlugin implements MethodApi<IEditor> {
+  private decoration: TextEditorDecorationType;
+  private diagnosticCollection: DiagnosticCollection;
+  constructor(options: EditorOptions) {
+    super(editorProfile);
+    super.setOptions(options);
+  }
+  setOptions(options: EditorOptions) {
+    super.setOptions(options);
+  }
+  onActivation() {
+    this.decoration = window.createTextEditorDecorationType({
+      backgroundColor: 'editor.lineHighlightBackground',
+      isWholeLine: true,
+    });
+    // @ts-ignore
+    this.diagnosticCollection = languages.createDiagnosticCollection(this.options.language);
+  }
+  onDeactivation() {
+    this.decoration.dispose();
+  }
+  async highlight(position: HighlightPosition, filePath: string, hexColor: string): Promise<void> {
+    const editors = window.visibleTextEditors;
+    // Parse `filePath` to ensure if a valid file path was supplied
+    const editor = editors.find(editor => editor.document.uri.path === Uri.parse(filePath).path);
+    if(editor) {
+      const start: Position = new Position(position.start.line, position.start.column);
+      const end: Position = new Position(position.end.line, position.end.column);
+      const newDecoration = { range: new Range(start, end) };
+      if (hexColor) {
+        this.decoration = window.createTextEditorDecorationType({
+          backgroundColor: hexColor,
+          isWholeLine: true,
+        });
+      }
+      return editor.setDecorations(this.decoration, [newDecoration]);
```

Comment:
If `setDecorations` returns void, then don't return it at all. Else update the output type of the method.